src/md/compiler/emit.cpp

```diff
@@ -1541,30 +1538,16 @@ STDMETHODIMP RegMeta::DefineUserString(       // S_OK or error.
     
     _ASSERTE(pstk && szString && cchString != ULONG_MAX);
 
-    //
-    // Walk the entire string looking for characters that would block us from doing
-    // a fast comparison of the string.  These characters include anything greater than
-    // 0x80 or an apostrophe or a hyphen.  Apostrophe and hyphen are excluded because
-    // they would prevent words like coop and co-op from sorting together in a culture-aware
-    // comparison.  We also need to exclude some set of the control characters.  This check
-    // is more restrictive 
-    //
-    for (i=0; i<cchString; i++) {
-        c = szString[i];
-        if (c>=0x80 || HighCharHelper::IsHighChar((int)c)) {
-            bIs80Plus = true;
-            break;
-        }
-    }
 
     // Copy over the string to memory.
     ulMemSize = cchString * sizeof(WCHAR);
     IfFailGo(qb.ReSizeNoThrow(ulMemSize + 1));
     pb = reinterpret_cast<PBYTE>(qb.Ptr());
     memcpy(pb, szString, ulMemSize);
     SwapStringLength((WCHAR *) pb, cchString);
-    // Set the last byte of memory to indicate whether there is a 80+ character.
-    *(pb + ulMemSize) = bIs80Plus ? 1 : 0;
+    // Always set the last byte of memory to indicate that there may be a 80+ or special character.
+    // This byte is not used by the runtime.
```

Comment:
Why set it at all now?

Reply:
We do not want to leave the memory uninitialized to avoid risk of unpredictable behavior. Setting it to 1 is the easy defensive choice.